src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -894,6 +929,34 @@ func (c controller) uploadSummaryLogs(podName string, namespace string, sinceSec
 	return nil
 }
 
+func (c controller) parseMustGatherImages() []string {
+	images := make([]string, 0)
+	if c.MustGatherImage == "" {
+		c.log.Infof("collecting must-gather logs into using image from release\n")
+		return images
+	}
+
+	var imageMap map[string]string
+	err := json.Unmarshal([]byte(c.MustGatherImage), &imageMap)
+	c.log.Infof("collecting must-gather logs into using a this image setup %s\n", c.MustGatherImage)
+	if err != nil {
+		images = append(images, c.MustGatherImage)
```

Comment:
Why would it be appended if there was an error?

Reply:
It means that the structure of the variable is the old one (non-JSON) and therefor can be taken as is

Reply:
In what scenario would it not be a JSON? is that backward-compatible? If so - please add a comment.

Reply:
Yes it is backward compatible - This is the part that takes card of the backward comparability. comment was added